src/controllers/contract/index.ts

```diff
@@ -0,0 +1,99 @@
+import { Request, Response } from 'express';
+import ContractService from '../../services/contract';
+import { CreateContractRequest, FindContractRequest, PatchContractRequest } from './types';
+import { handlePrismaError } from '../../lib/handle-prisma-error';
+import UserService from '../../services/user';
+import ContractCategoryService from '../../services/contractCategories';
+
+export default class ContractController {
+  static async getAll(_: Request, response: Response) {
+    const contracts = await ContractService.getAll();
+    return response.success({ data: contracts });
+  }
+
+  static async find(request: Request, response: Response) {
+    const { id } = request.params as any as FindContractRequest;
+
+    const contract = await ContractService.find({ id });
+
+    if (!contract) {
+      return response.notfound();
+    }
+
+    return response.success({ data: contract });
+  }
+
+  static async create(request: Request, response: Response) {
+    const body = request.body as any as CreateContractRequest;
+
+    if (!body.beneficiary_id) {
+      return response.badrequest({ errors: { beneficiary_id: 'Beneficiary is required ' } });
+    }
+    if (!body.provider_id) {
+      return response.badrequest({ errors: { provider_id: 'Provider is required' } });
+    }
+    if (!body.category_id) {
+      return response.badrequest({ errors: { category_id: 'Category is required' } });
+    }
+    if (!body.start_date) {
+      return response.badrequest({ errors: { start_date: 'Start date is required' } });
+    }
+    if (!body.end_date) {
+      return response.badrequest({ errors: { end_date: 'End date is required' } });
+    }
```

Comment:
if end_date is not necessary, then why are we validating and saying it is required?